### ISSUE TITLE
Revert icon

### ISSUE THREAD

**Issue by hockeymikey** [y20k/trackbook#54]
The new blue app icon is super ugly and hard to look at. You should revert to the old red one which was perfect and has a pleasure to use.

**Comment by y20k** [comment]
Hi @hockeymikey 

I changed the icon anticipating a [change](https://www.xda-developers.com/googles-new-requirements-will-make-every-play-store-icon-a-squircle/) in the Play Store. Also some launchers force app icons to be a certain shape. It is recommended to use so called [Adaptive Icons](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive) to fit that need. I tried to convert the old icon to an adaptive icon, but it’s shape was to complex for an direct translation into an adaptive icon. 

So what I did was: I stripped down the book icon to two open book pages without the cover and I put that onto a plain blue background that can have many shapes. I agree that this change may not be ideal. Or. I am not sure if the feeling is normal because people including me often have a hard time with new icons.

Anyway. Do you have another idea than just reverting? As I wrote the old icons needed to be adapted. And simply putting it onto a say grey background was not pretty, too.

**Comment by Hob111** [comment]
I might have known, the reason that many naff icons have started to appear on my devices is down to Google! Presumably apps distributed solely through F-Droid won't suffer from this?

Personally I don't like any of the tiny icons in a circle of colour but it seems we're stuck with them, I prefer ones that "float" on the background screen image. I can't suggest anything better if, as you say, the original icon when adapted to fit the new format didn't work, except maybe changing the background to white or black.

**Comment by y20k** [comment]
Actually there are two problems. 

1. Play Store 
2. Launchers / Home Screens 

The later begin to enforce rounded icons too. If anyone is interested in dabbling around, I put up the SVG file for the logo up here a while ago.

https://github.com/y20k/trackbook/blob/master/assets/trackbook-app-icon-current.svg

PS. The blue in the icon is the color used in the user interface of Trackbook.

**Comment by Hob111** [comment]
Ah, that explains why my latest phone (Android 8) is so different to my last one (Android 6). It wouldn't even let me install my favourite launcher (Catapult).

White is also used in the user interface  :wink:.  Seriously though, I was just trying to think of something that might make the icon more pleasing to @hockeymikey and presumably others too. The only other thing I can suggest is to produce an icon that looks just like the start "button" in Trackbook, the minimalist approach.

**Comment by hockeymikey** [comment]
@y20k God, everyday I hate Google more and more. Their radioactive decisions that make for the Android platform are the worst. 
@Hob111 Here I will personally disagree, the minimalist icon would be even worse. I liked the red better, but it also had more pop before. This current icon seems flat and too small. 
Not sure if it's an option to include multiple icons and if launchers like Nova and the like can distinguish between them giving users the ability to customize how they like.

**Comment by y20k** [comment]
Just for the discussion here: I tried out a few colors for the background. 

### On Black Background

![icons](https://user-images.githubusercontent.com/9103935/56650978-6524b280-6688-11e9-92e2-15053b3b2805.png)

### Larger Versions On Click

<a href="https://user-images.githubusercontent.com/9103935/56642329-7d400600-6677-11e9-8862-7fafe0319c37.png"><img src="https://user-images.githubusercontent.com/9103935/56642329-7d400600-6677-11e9-8862-7fafe0319c37.png" width="36"></a>

<a href="https://user-images.githubusercontent.com/9103935/56642330-7d400600-6677-11e9-96c1-9c3a24e8786a.png"><img src="https://user-images.githubusercontent.com/9103935/56642330-7d400600-6677-11e9-96c1-9c3a24e8786a.png" width="36"></a>

<a href="https://user-images.githubusercontent.com/9103935/56642443-b8dad000-6677-11e9-973c-ab3822e00b0f.png"><img src="https://user-images.githubusercontent.com/9103935/56642443-b8dad000-6677-11e9-973c-ab3822e00b0f.png" width="36"></a>

**Comment by Hob111** [comment]
@hockeymikey My suggestion was intended to be tongue in cheek, maybe I should have added a :smile:. It seems that icons on my phone are becoming little more than one circle of colour inside another.
@y20k If we're voting I prefer the red.

**Comment by y20k** [comment]
> If we're voting I prefer the red.

My wife does too :wink:

**Comment by Hob111** [comment]
It must be my feminine side!  :woman:

**Comment by hockeymikey** [comment]
I think it's because the red makes the icon pop much better. Has alot of green in it and the gray which is more of a cool color. The blue makes it sort of blend together.

**Comment by Hob111** [comment]
@y20k As a matter of interest, do you know what will happen if an app doesn't change it's icon to be compliant? I use a number of apps which don't have adaptive icons. Some are stable apps that haven't been updated for a number of years and may never be, others, like F-Droid, aren't likely to be a problem on Play Store :smirk: but what about launchers / home screens?

**Comment by y20k** [comment]
@Hob111 
Adaptive Icons (icon that can have multiple shapes) have been introduced two years ago, and it seems that Google is slowly beginning to nudge developers to use them. Not only in the Play Store, but also in the ASOP launcher. I believe that other OEM will follow.

I found the following quote on Reddit

> In addition, starting in O, AOSP Launcher3 code will be automatically stamping icons of apps that are targeting O but that haven’t updated to an AdaptiveIcon. In the future, as adoption grows, we will also consider stamping legacy icons more aggressively, ie. for every app, regardless of their target SDK.

Source: [AndroidEngTeam AMA on Reddit](https://www.reddit.com/r/androiddev/comments/6nuq0a/were_on_the_engineering_team_for_android_o_ask_us/dkfxebs/) (via [Medium](https://medium.com/google-design/understanding-android-adaptive-icons-cee8a9de93e2))

So I guess that Adaptive Icons are here to stay for a while. I am not happy, too. I am not a designer. I liked my old icon. The street corner on the icon is the place where I lived through my childhood. So I guess I would like to stay with the open pages and this map snippet.

@hockeymikey & @Hob111 
One cannot too often change a product icon, but I think I will change the background with the next release. Any other suggestion than red?

**Comment by y20k** [comment]
Okay. Then lets got with red. I tried several background variations: 

1. plain red --> a bit boring (see my comment above)
2. red color gradient --> makes the icon look washed out
3. plain red with a crease/kink --> a good compromise - I like it best so far

### Adaptive Icon with red background 

Version: plain red with a crease in the middle

<a href="https://user-images.githubusercontent.com/9103935/57179707-d9c5d100-6e80-11e9-8b1e-aae16acecb39.png"><img src="https://user-images.githubusercontent.com/9103935/57179707-d9c5d100-6e80-11e9-8b1e-aae16acecb39.png" width="60" /><a>

<a href="https://user-images.githubusercontent.com/9103935/57179706-d9c5d100-6e80-11e9-9456-df72a4685cb2.png"><img src="https://user-images.githubusercontent.com/9103935/57179706-d9c5d100-6e80-11e9-9456-df72a4685cb2.png" width="60" /><a>

**Comment by Hob111** [comment]
The crease is a nice touch, it integrates the icon. 

On the need for this change, I liked the comment on the Reddit link you posted above.

> I just love how removing custom icon shapes is called "giving more control of their icons" in pr-speak.

**Comment by y20k** [comment]
Yesterday I came across an older screenshot of my home screen - featuring the classic Trackbook icon. I must say I liked the older icon better. :cry: 

<a href="https://raw.githubusercontent.com/y20k/trackbook/v1.1.17/assets/trackbook-app-icon-current.png"><img src="https://raw.githubusercontent.com/y20k/trackbook/v1.1.17/assets/trackbook-app-icon-current.png" width="72" /></a>

<a href="https://user-images.githubusercontent.com/9103935/57179706-d9c5d100-6e80-11e9-9456-df72a4685cb2.png"><img src="https://user-images.githubusercontent.com/9103935/57179706-d9c5d100-6e80-11e9-9456-df72a4685cb2.png" width="72" /><a>

But as I wrote, adaptive icons are here to stay.

Here a screenshot of the current stock home app on Lineage OS: Trebuchet. Here one can see how launchers handle non-adaptive icons.
<a href="https://user-images.githubusercontent.com/9103935/57480687-b778ec80-72a0-11e9-8c8c-22869505e35d.png"><img src="https://user-images.githubusercontent.com/9103935/57480687-b778ec80-72a0-11e9-8c8c-22869505e35d.png" width="300" /></a>

Anyway. I think the new icons needs still to be tweaked a bit. I was "living" with the new icon for a week or so. I think that the red background needs to be toned down a bit. Another idea would to bring back the book cover that I removed earlier in order to simplify. (I must say designing icons is really not my strongest talent.)

<a href="https://user-images.githubusercontent.com/9103935/57179706-d9c5d100-6e80-11e9-9456-df72a4685cb2.png"><img src="https://user-images.githubusercontent.com/9103935/57179706-d9c5d100-6e80-11e9-9456-df72a4685cb2.png" width="72" /><a> <a href="https://user-images.githubusercontent.com/9103935/57481541-a4ffb280-72a2-11e9-93a8-c5b22d5f6329.png"><img src="https://user-images.githubusercontent.com/9103935/57481541-a4ffb280-72a2-11e9-93a8-c5b22d5f6329.png" width="72" /><a>

<a href="https://user-images.githubusercontent.com/9103935/57481755-21929100-72a3-11e9-959a-3e4c520635c1.png"><img src="https://user-images.githubusercontent.com/9103935/57481755-21929100-72a3-11e9-959a-3e4c520635c1.png" width="300" /></a>

**Any thoughts?**

**Comment by y20k** [comment]
If anyone is interested how the new icon looks in real life. I compiled a preview release of the next version of Trackbook.

* [trackbook-app-testing-1.2.1-001.apk](https://drive.google.com/file/d/1rPqZNBvRgmIddaeD6ksZzHzmCZ0Akofa/view?usp=sharing)

**Comment by y20k** [comment]
Okay. It's me again. Icon time again. I continue to be unsatisfied with the current icon. Today I tried to alter the color scheme a bit. 

1. Classic Color Scheme
2. Monochrome Colors
3. Lighter Colors

<a href="https://raw.githubusercontent.com/y20k/trackbook/87d3c955b46f7522d77258e24fdfe504bdd0dc34/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png"><img src="https://raw.githubusercontent.com/y20k/trackbook/87d3c955b46f7522d77258e24fdfe504bdd0dc34/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png" width="96" /></a><a href="https://raw.githubusercontent.com/y20k/trackbook/3e12ddaa4626684e83aac901a9d7ad750ae65667/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png"><img src="https://raw.githubusercontent.com/y20k/trackbook/3e12ddaa4626684e83aac901a9d7ad750ae65667/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png" width="96" /></a><a href="https://raw.githubusercontent.com/y20k/trackbook/eb18d7013f6bf68aefbca9c162400e04cf2cacab/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png"><img src="https://raw.githubusercontent.com/y20k/trackbook/eb18d7013f6bf68aefbca9c162400e04cf2cacab/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png" width="96" /></a>

... and here a test APK with the "Lighter Colors" variant of the logo. I prefer this one - especially on my actual home screen.

* [trackbook-app-testing-1.2.1-002.apk](https://drive.google.com/file/d/1r0O7R8Li_2BndIZZQehwWMTx9WnAPcGM/view?usp=sharing)

**Comment by Hob111** [comment]
@y20k The return of the cover is welcome and I think the lighter colours look better. Two questions. Why did you make the cover blue when it was previously white? Could it be toned down a bit, say to the shade in the monochrome version?

**Comment by y20k** [comment]
@Hob111 Blue? The book cover is supposed to be a dark grey. But I must admit I often get grey and other colors confused - to the point that my personal environment makes fun of me :-)

Anyway I will try the toned down book cover color you suggested.

**Comment by Hob111** [comment]
@y20k I think on this occasion you are perfectly correct and my laptop screen viewed from a less than ideal angle is to blame - or maybe it's me to blame for not looking from an ideal angle 😄 . It certainly looks dark grey on my phone screen and I think the lighter grey would be better balanced. 
However, when I look more closely I notice the monochrome image has an even lighter grey outline to the book pages which creates an impression of depth. This seems to be missing from the "Lighter Colors" variant of the logo and I wonder if it's this, as much as the lighter grey cover, which attracts my eye?
Maybe you should ask your wife? 😉

**Comment by y20k** [comment]
Two more screenshots with icon variations:


Variant 1: with the book cover color you preferred
<a href="https://user-images.githubusercontent.com/9103935/58477468-e540ab00-8153-11e9-868e-664e90006f6a.png"><img src="https://user-images.githubusercontent.com/9103935/58477468-e540ab00-8153-11e9-868e-664e90006f6a.png" width="300" /></a>

Variant 2: on grey background
<a href="https://user-images.githubusercontent.com/9103935/58477477-ea9df580-8153-11e9-92c5-77730f073f03.png"><img src="https://user-images.githubusercontent.com/9103935/58477477-ea9df580-8153-11e9-92c5-77730f073f03.png" width="300" /></a>

I must say I am a bit lost. Ultimately I think the icon is not really suited to be an Adaptive Icon. So the goal for now can only be, make it as good as possible. In the long term the app would need a completely new icon.

**Comment by y20k** [comment]
PS. I like variant 1 the best for now.
PPS. I will probably ask my wife.

**Comment by Hob111** [comment]
I think you're doing OK. I like both Variant 1 and 2 but note that neither show the lighter outline to the book pages that I can see in the monochrome icon above. Is it possible to add it to variants 1 and 2? It enhances the 3D effect created by the central "fold".

**Comment by y20k** [comment]
Thanks @Hob111 for insisting on the lighter gray. I confused two gray variants for the book cover. Anyway. In the latest iteration I made the book cover a lot lighter. I think now the icon and the background blend better, without blurring into each other - as was the case with the blue icon.

I also made the left side of the background darker. While I like the effect I think the background draws a bit too much attention in that variant.

### Regular Fold Effect
<a href="https://user-images.githubusercontent.com/9103935/58728834-11745a00-83e8-11e9-8d7f-6116167dea7a.png"><img src="https://user-images.githubusercontent.com/9103935/58728834-11745a00-83e8-11e9-8d7f-6116167dea7a.png" width="256" /></a>

<a href="https://user-images.githubusercontent.com/9103935/58728854-1cc78580-83e8-11e9-848a-26e4c9bcbabf.png"><img src="https://user-images.githubusercontent.com/9103935/58728854-1cc78580-83e8-11e9-848a-26e4c9bcbabf.png" width="300" /></a>


### Stronger Fold Effect
<a href="https://user-images.githubusercontent.com/9103935/58728845-176a3b00-83e8-11e9-8625-6373343f2074.png"><img src="https://user-images.githubusercontent.com/9103935/58728845-176a3b00-83e8-11e9-8625-6373343f2074.png" width="256" /></a>

<a href="https://user-images.githubusercontent.com/9103935/58728850-1a652b80-83e8-11e9-936f-35c9393c0a98.png"><img src="https://user-images.githubusercontent.com/9103935/58728850-1a652b80-83e8-11e9-936f-35c9393c0a98.png" width="300" /></a>

**Comment by Hob111** [comment]
I think you've produced a pretty good icon, your pessimism of a few post ago was unwarranted. The "regular" icon looks best imo, the other is a bit too much.

**Comment by y20k** [comment]
I did do a few tests in different sizes and I finally decided to go with this variation. 

<a href="https://user-images.githubusercontent.com/9103935/58946245-be066100-8785-11e9-9971-eeae654e45ef.png"><img src="https://user-images.githubusercontent.com/9103935/58946245-be066100-8785-11e9-9971-eeae654e45ef.png" width="256"></a>


The update should be on F-Droid in the next couple of days. Thanks for helping me out. And thanks for insisting on a change. I like the red icon better now than the blue one.
